panic: runtime error: index out of range

I'm not sure if this is a bug. Our fork is running the `fix-seek` branch from earlier today. Keys are < 30 bytes, values are < 10 bytes.

It's going to be harder for me to provide an example program demonstrating this since I'm swapping out LevelDB in something. Hopefully I'm doing something obviously wrong?

```
panic: runtime error: index out of range

goroutine 19 [running]:
runtime.panic(0x4470b00, 0x4a33197)
    /usr/local/Cellar/go/1.2.2/libexec/src/pkg/runtime/panic.c:266 +0xb6
github.com/VividCortex/bolt.(*node).write(0xc2161beb60, 0xc23432d000)
    /Users/preetamjinka/git/influxdb/src/github.com/VividCortex/bolt/node.go:185 +0x3ab
github.com/VividCortex/bolt.(*node).spill(0xc217e42930, 0xc229044480, 0x4ccfa60)
    /Users/preetamjinka/git/influxdb/src/github.com/VividCortex/bolt/node.go:320 +0x372
github.com/VividCortex/bolt.(*Bucket).spill(0xc229044450, 0xc229041500, 0x4)
    /Users/preetamjinka/git/influxdb/src/github.com/VividCortex/bolt/bucket.go:543 +0x182
github.com/VividCortex/bolt.(*Bucket).spill(0xc22afd4018, 0x743e899, 0x4a38980)
    /Users/preetamjinka/git/influxdb/src/github.com/VividCortex/bolt/bucket.go:514 +0x7bb
github.com/VividCortex/bolt.(*Tx).Commit(0xc22afd4000, 0x0, 0x0)
    /Users/preetamjinka/git/influxdb/src/github.com/VividCortex/bolt/tx.go:159 +0x1a9
github.com/VividCortex/bolt.(*DB).Update(0xc228fa3c80, 0x4ccfe08, 0xc2100d4e86, 0x7)
    /Users/preetamjinka/git/influxdb/src/github.com/VividCortex/bolt/db.go:464 +0xe7
```

--- a/page.go
+++ b/page.go
@@ -10,7 +10,6 @@ const pageHeaderSize = int(unsafe.Offsetof(((*page)(nil)).ptr))
 
 const maxAllocSize = 0xFFFFFFF
 const minKeysPerPage = 2
-const maxNodesPerPage = 65535
 
 const branchPageElementSize = int(unsafe.Sizeof(branchPageElement{}))
 const leafPageElementSize = int(unsafe.Sizeof(leafPageElement{}))
@@ -57,23 +56,23 @@ func (p *page) meta() *meta {
 
 // leafPageElement retrieves the leaf node by index
 func (p *page) leafPageElement(index uint16) *leafPageElement {
-	n := &((*[maxNodesPerPage]leafPageElement)(unsafe.Pointer(&p.ptr)))[index]
+	n := &((*[0xFFFFFFF]leafPageElement)(unsafe.Pointer(&p.ptr)))[index]
 	return n
 }
 
 // leafPageElements retrieves a list of leaf nodes.
 func (p *page) leafPageElements() []leafPageElement {
-	return ((*[maxNodesPerPage]leafPageElement)(unsafe.Pointer(&p.ptr)))[:]
+	return ((*[0xFFFFFFF]leafPageElement)(unsafe.Pointer(&p.ptr)))[:]
 }
 
 // branchPageElement retrieves the branch node by index
 func (p *page) branchPageElement(index uint16) *branchPageElement {
-	return &((*[maxNodesPerPage]branchPageElement)(unsafe.Pointer(&p.ptr)))[index]
+	return &((*[0xFFFFFFF]branchPageElement)(unsafe.Pointer(&p.ptr)))[index]
 }
 
 // branchPageElements retrieves a list of branch nodes.
 func (p *page) branchPageElements() []branchPageElement {
-	return ((*[maxNodesPerPage]branchPageElement)(unsafe.Pointer(&p.ptr)))[:]
+	return ((*[0xFFFFFFF]branchPageElement)(unsafe.Pointer(&p.ptr)))[:]
 }
 
 // dump writes n bytes of the page to STDERR as hex output.

